transmart-api-server/src/main/groovy/org/transmartproject/api/server/client/CustomClientRequestFactory.groovy

```diff
@@ -0,0 +1,64 @@
+package org.transmartproject.api.server.client
+
+import org.apache.http.client.methods.HttpUriRequest
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.beans.factory.config.ConfigurableBeanFactory
+import org.springframework.context.annotation.Scope
+import org.springframework.http.*
+import org.springframework.http.client.ClientHttpRequestFactory
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory
+import org.springframework.stereotype.Component
+import org.springframework.util.LinkedMultiValueMap
+import org.springframework.util.MultiValueMap
+import org.springframework.web.client.RestTemplate
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+class CustomClientRequestFactory extends HttpComponentsClientHttpRequestFactory implements ClientHttpRequestFactory {
```

Comment:
Ah. I was confused with naming above. Could it be renamed to `OfflineTokenRequestFactory` or smth.?